service/src/main/scala/za/co/absa/loginsvc/rest/config/provider/ConfigProvider.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.config.provider
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import pureconfig._
+import pureconfig.generic.auto._
+import pureconfig.module.yaml._
+import za.co.absa.loginsvc.rest.config.actuator.GitConfig
+import za.co.absa.loginsvc.rest.config.auth._
+import za.co.absa.loginsvc.rest.config.validation.ConfigValidationException
+import za.co.absa.loginsvc.rest.config.{BaseConfig, JwtConfig}
+
+import java.nio.file.{InvalidPathException, Paths}
+
+@Component
+class ConfigProvider(@Value("${DefaultYamlPath:service/src/main/resources/application.yaml}") yamlContent: String)
+  extends JwtConfigProvider
+    with AuthConfigProvider {
+
+  private var yamlConfig: YamlConfigSource = {
+    if (isValidPath(yamlContent))
+      YamlConfigSource.file(yamlContent)
+    else
+      YamlConfigSource.string(yamlContent)
```

Comment:
What is the point of this? Why do expect, in the case of the path being invalid, to find a usable yaml definition in the very same string?

Reply:
Removed. This was left over from an older implementation